### PR TITLE
App Submission: Torq (https://github.com/lncapital/torq)

### LINKED ISSUE #209: App request: Torq
Have not used it myself, but heard of this. Looks similar to LNDg, but with a better UI. 

Sent to the Umbrel Twitter account, was suggested that I share here as well. 

https://github.com/lncapital/torq

### PR DESCRIPTION
**Pull request by maxwedwards**
# App Submission

### App name
torq

Fixes #209 

### 256x256 SVG icon
https://svgshare.com/i/nyJ.svg

### Gallery images
forwards-summary
forwards-table
Inspect channel
payments
Torq-Flow

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [X] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [X] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)
- [X] [Umbrel community App Store](https://github.com/lncapital/umbrel-community-app-store)

I've tried to use the dev environment but it asked me to install Pro or Business of Parallels so I stopped

### CODE REVIEW

**Comment by maxwedwards** [comment]
Let us know if you need any help reviewing our submission. We would love to be in the app store.

**Review by nevets963 — COMMENTED** [review]
@maxwedwards Many thanks for submitting Torq into the App Store!

Would you be able to grant me write access to this branch, so I can make a few changes?

**Comment by ohenrik** [comment]
> @maxwedwards Many thanks for submitting Torq into the App Store!
> 
> 
> 
> Would you be able to grant me write access to this branch, so I can make a few changes?

I have added you with write access now ⚡️

File: torq/docker-compose.yml
Comment (nevets963) [inline torq/docker-compose.yml:24]: Is it possible to remove the password authentication? Umbrel automatically protects apps from unauthorised access using the user's credentials (via the App Proxy).

File: torq/docker-compose.yml
Comment (nevets963) [inline torq/docker-compose.yml:26]: During testing I had a small issue with my LND instance. Torq would get stuck in a reboot loop with this error:

```
2022/11/20 13:05:36 WARNING: [core] [Channel #1 SubChannel #2] grpc: addrConn.createTransport failed to connect to {
  "Addr": "10.21.21.9:10009",
  "ServerName": "10.21.21.9:10009",
  "Attributes": null,
  "BalancerAttributes": null,
  "Type": 0,
  "Metadata": null
}. Err: connection error: desc = "transport: error while dialing: dial tcp 10.21.21.9:10009: connect: no route to host"
{"level":"error","error":"Getting public key from node: Can't connect to node to verify public key, check all details including TLS Cert and Macaroon: cannot dial to lnd: connection error: desc = \"transport: error while dialing: dial tcp 10.21.21.9:10009: connect: no route to host\"","time":"2022-11-20T13:05:36Z","message":"Adding node specified in config to database"}
{"level":"fatal","error":"Adding node specified in config to database: Getting public key from node: Can't connect to node to verify public key, check all details including TLS Cert and Macaroon: cannot dial to lnd: connection error: desc = \"transport: error while dialing: dial tcp 10.21.21.9:10009: connect: no route to host\"","time":"2022-11-20T13:05:36Z"}
```

This is understandable, however, it would be nicer user experience, if the container remained up and displayed this error in the browser. I'm not going to block this PR on this - just a suggestion.

**Review by nevets963 — APPROVED** [review]
@maxwedwards @ohenrik Many many thanks for the PR to add Torq into the Umbrel App Store!!

I've pushed a few commits to simplify the PR as well as included a `passwd` file to ensure that hosts that do not have a user with a UID 1000 continue to work.
I also noted that the Docker image digests were for a specific platform. You can get multi-arch digests by running `docker pull <image>` and it will display the digest at the bottom.

I've left 2 questions below - which are non-blocking for this PR.

Tested on `arm64` and `amd64`.

Approved from me!

@mayankchhabra Can you help us with the app store graphics so we get this launched 🚀

File: torq/docker-compose.yml
Comment (maxwedwards) [inline torq/docker-compose.yml:42]: Great suggestion thanks

File: torq/docker-compose.yml
Comment (maxwedwards) [reply torq/docker-compose.yml:24]: As it stands it's not possible for us to run passwordless but we will add this in shortly for a nicer experience on Umbrel. Would be be ok to send a PR in later removing password requirement?

File: torq/docker-compose.yml
Comment (maxwedwards) [reply torq/docker-compose.yml:26]: Thanks for the feedback. We have recently done some work to retry the connection and to not prevent the webserver from coming up in this situation which should be a marginally better experience. Showing the current connection status (with any associated errors) is something we will be looking into very shortly. Much requested!

File: torq/docker-compose.yml
Comment (nevets963) [reply torq/docker-compose.yml:24]: Yes absolutely 👍

**Comment by maxwedwards** [comment]
Thanks for all your amendments @nevets963 and for the tip on the digests, I couldn't figure out how to get the multi-arch one! As far as I know there's nothing more you need from us right now? We will update Torq with your suggestions but hope to open a new PR with a version bump.

**Comment by nevets963** [comment]
@maxwedwards Everything is good from me. We only now need graphics for the app store to be pushed here: https://github.com/getumbrel/umbrel-apps-gallery which @mayankchhabra can help us with. Once those graphics are merged in, I'll merge in this and then it will appear within Umbrel 😄

**Comment by maxwedwards** [comment]
@nevets963 we have a more recent tag, are you happy for me to update this PR with it or would you like us to wait until after this is merged? And how much work is it for you to look through our PRs with version bumps? We do release fairly often so we don't want to cause you a lot of work and could perhaps just update Umbrel monthly. What is a reasonable cadence for you?

**Comment by nevets963** [comment]
@maxwedwards sure that’s fine. I can do a re-test on my side. Monthly releases are totally fine, we have a few apps that release roughly every 2 weeks.

**Comment by maxwedwards** [comment]
@nevets963 bumped to v0.11.1

**Review by nevets963 — APPROVED** [review]
Torq 0.11.1 tested on `arm64` and `amd64`.

**Comment by maxwedwards** [comment]
Yep, all good with us!

**Comment by nevets963** [comment]
Everything looks geat!
Screenshot 2022-11-22 at 21 27 49

"Houston, we are good for launch" 🚀

**Comment by maxwedwards** [comment]
Amazing thanks!

**Comment by maxwedwards** [comment]
@nevets963 the Github pages version of the app gallery doesn't seem to have updated yet.

Getting 404 on: https://getumbrel.github.io/umbrel-apps-gallery/torq/2.png

**Comment by maxwedwards** [comment]
Ah, it looks like it should be `2.jpg` not `2.png`

**Comment by nevets963** [comment]
@maxwedwards I can confirm the assets are live. I fixed the filenames in this commit: https://github.com/getumbrel/umbrel-apps/pull/269/commits/126316529584c8d5d29e5d635d07639ad6f3373d

**Pull request merged**
